### PR TITLE
42389 uikit rebase

### PR DESCRIPTION
**Pull request by rikki-iki**

### CODE REVIEW

File: agov/modules/custom/text_icon/src/Plugin/Field/FieldFormatter/TextIconFormatter.php [hunk 1]
```diff
@@ -20,15 +21,60 @@
...
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $element = array();
+
+    foreach ($items as $delta => $item) {
+      $element[$delta] = array(
+        '#type' => 'markup',
+        '#markup' => '<svg role="presentation"><use xlink:href="' . $this->getSetting('icon_sprite') . '#' . $item->value . '" /></svg>',
```
Comment (rikki-iki) [inline agov/modules/custom/text_icon/src/Plugin/Field/FieldFormatter/TextIconFormatter.php]: @benjy this markup doesn't print, I'm guessing it's being sanitised? Is there a way around it?

File: agov/modules/custom/agov_default_content/content/block_content/69e9555f-f0f3-4dd6-b905-5b1efffa120f.json [hunk 2]
```diff
@@ -50,23 +50,39 @@
     "field_social_icon": [
         {
             "uri": "http:\/\/www.facebook.com\/pages\/PreviousNext\/170975076260877",
-            "title": "Facebook",
-            "options": []
+            "title": "Like us on Facebook",
+            "options": [
+                {
+                    "class": "social--facebook"
```
Comment (rikki-iki) [inline agov/modules/custom/agov_default_content/content/block_content/69e9555f-f0f3-4dd6-b905-5b1efffa120f.json]: @benjy What's the right syntax here? The classes don't get added on a flesh install but not sure how to write it

File: agov/modules/custom/text_icon/src/Plugin/Field/FieldFormatter/TextIconFormatter.php (hunk 1 above)
Comment (benjy) [inline agov/modules/custom/text_icon/src/Plugin/Field/FieldFormatter/TextIconFormatter.php]: I think you'll want FormattableMarkup here, and the user input will be an argument that gets filtered.

File: agov/modules/custom/agov_default_content/content/block_content/69e9555f-f0f3-4dd6-b905-5b1efffa120f.json (hunk 2 above)
Comment (benjy) [inline agov/modules/custom/agov_default_content/content/block_content/69e9555f-f0f3-4dd6-b905-5b1efffa120f.json]: Are you manually changing the default content? You should be able to install it, change in the UI and then export it back?

File: agov/modules/custom/text_icon/src/Plugin/Field/FieldFormatter/TextIconFormatter.php [hunk 3]
```diff
@@ -20,15 +21,60 @@
...
  */
-class TextIconFormatter extends StringFormatter {
+class TextIconFormatter extends FormatterBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    return array(
+      'icon_sprite' => '/profiles/agov/themes/contrib/agov_base/sass/base/icons/sprite--content.svg',
```
Comment (benjy) [inline agov/modules/custom/text_icon/src/Plugin/Field/FieldFormatter/TextIconFormatter.php]: We can't do this, it won't work for everyone.

Comment (rikki-iki) [inline agov/modules/custom/text_icon/src/Plugin/Field/FieldFormatter/TextIconFormatter.php]: It's editable though

File: agov/modules/custom/agov_default_content/content/block_content/69e9555f-f0f3-4dd6-b905-5b1efffa120f.json (hunk 2 above)
Comment (rikki-iki) [inline agov/modules/custom/agov_default_content/content/block_content/69e9555f-f0f3-4dd6-b905-5b1efffa120f.json]: I did `make config-export` but it didn't change this?

File: agov/modules/custom/text_icon/src/Plugin/Field/FieldFormatter/TextIconFormatter.php (hunk 3 above)
Comment (benjy) [inline agov/modules/custom/text_icon/src/Plugin/Field/FieldFormatter/TextIconFormatter.php]: Lets just leave it empty by default then and have our own agov config set it to a custom value?

File: agov/modules/custom/text_icon/src/Plugin/Field/FieldFormatter/TextIconFormatter.php (hunk 1 above)
Comment (rikki-iki) [inline agov/modules/custom/text_icon/src/Plugin/Field/FieldFormatter/TextIconFormatter.php]: FormattableMarkup worked thanks

**Comment by rikki-iki** [comment]
test this please

**Comment by rikki-iki** [comment]
@benjy do you mind checking why this one is failing? Something about field_icon_block_icon settings missing schema?

**Comment by benjy** [comment]
It's probably the icon_sprite setting, that's new?

**Comment by benjy** [comment]
We'll need to create agov/modules/text_icon/config/schema/text_icon.schema.yml with appropriate schema. Checkout another formatter for an example or ping me for some help.

**Comment by benjy** [comment]
LGTM

File: agov/modules/custom/text_icon/src/Plugin/Field/FieldFormatter/TextIconFormatter.php [hunk 4]
```diff
@@ -20,15 +21,63 @@
...
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $elements = parent::settingsForm($form, $form_state);
+
+    $elements['icon_sprite'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Icon sprite location'),
```
Comment (rikki-iki) [inline agov/modules/custom/text_icon/src/Plugin/Field/FieldFormatter/TextIconFormatter.php]: Fixed! Curious what the difference it?

**Pull request merged**